src/common/bots/Bot.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef Bot_hpp
+#define Bot_hpp
+
+#include "../../common/objects/Mine.hpp"
+#include "../../common/util/InputDef.hpp"
+#include "../../common/util/Util.hpp"
+#include "../../common/util/game_settings.hpp"
+
+#include <cstdlib>
+#include <ctime>
+#include <unistd.h>
+
+#include <iostream>
+#include <memory>
+
+enum BotStrategy { Random, NearestGreedy, MostNeededGreedy, Groupie, GreedieGroupie };
+
+class Bot {
+  public:
+    Bot();
+    ~Bot(){};
+    Bot(const Bot& temp_obj) = delete;
+    Bot& operator=(const Bot& temp_obj) = delete;
+
+    std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
```

Comment:
maybe we can make a typedef for `std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>` in `InputDef.hpp`

Reply:
Im  actually wary of adding abstraction just because we can. it means one more lookup everytime you see "Inputs" or w/e you end up calling it